python/paddle/v2/fluid/tests/book/test_fit_a_line.py

```diff
@@ -14,28 +16,46 @@
 sgd_optimizer = fluid.optimizer.SGD(learning_rate=0.001)
 sgd_optimizer.minimize(avg_cost)
 
-BATCH_SIZE = 20
-
 train_reader = paddle.batch(
     paddle.reader.shuffle(
         paddle.dataset.uci_housing.train(), buf_size=500),
     batch_size=BATCH_SIZE)
 
+test_reader = paddle.batch(
+    paddle.dataset.uci_housing.test(),
+    batch_size=128)
+
 place = fluid.CPUPlace()
-feeder = fluid.DataFeeder(place=place, feed_list=[x, y])
 exe = fluid.Executor(place)
+feeder = fluid.DataFeeder(place=place, feed_list=[x, y])
 
 exe.run(fluid.default_startup_program())
 
 PASS_NUM = 100
-for pass_id in range(PASS_NUM):
-    fluid.io.save_persistables(exe, "./fit_a_line.model/")
-    fluid.io.load_persistables(exe, "./fit_a_line.model/")
-    for data in train_reader():
-        avg_loss_value, = exe.run(fluid.default_main_program(),
-                                  feed=feeder.feed(data),
-                                  fetch_list=[avg_cost])
-
-        if avg_loss_value[0] < 10.0:
-            exit(0)  # if avg cost less than 10.0, we think our code is good.
-exit(1)
+
+
+def train():
+    for pass_id in range(PASS_NUM):
+        fluid.io.save_persistables(exe, "./fit_a_line.model/")
+        fluid.io.load_persistables(exe, "./fit_a_line.model/")
+        for data in train_reader():
+            avg_loss_value,  = exe.run(fluid.default_main_program(),
+                                       feed=feeder.feed(data),
+                                       fetch_list=[avg_cost])
+
+            if avg_loss_value[0] < 10.0:
+                return  # if avg cost less than 10.0, we think our code is good.
+
+
+print("Training...")
+train()
+
+print("Now performing inference...")
+fluid.io.load_persistables(exe, "./fit_a_line.model/")
+for data in test_reader():
+    out, y_pred, y_label = exe.run(fluid.default_main_program(),
```

Comment:
You must clone the main_program first.

Reply:
Please refer to https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/v2/fluid/tests/book/test_recognize_digits_mlp.py#L35.
We need to create a inference program first.